packages/core/src/releases.ts

```diff
@@ -0,0 +1,577 @@
+import compareVersions from 'compare-versions'
+import regexEscape from 'escape-string-regexp'
+
+import { getVersionInfo } from './versions.js'
+import { transformTemplate } from './transform-template.js'
+import {
+	CommitWithAssociatedPullRequests,
+	GitHubRelease,
+	Label,
+	PullRequest,
+	ReleaseDrafterCategorizedPullRequest,
+	ReleaseDrafterCategory,
+	ReleaseDrafterConfig,
+	ReleaseDrafterContext,
+	VersionResolver,
+} from './types.js'
+import { ReleaseType } from 'semver'
+
+const sortReleases = (releases: GitHubRelease[]) => {
+	// For semver, we find the greatest release number
+	// For non-semver, we use the most recently merged
+	try {
+		return releases.sort((r1, r2) => compareVersions(r1.tag_name, r2.tag_name))
+	} catch {
+		return releases.sort(
+			(r1, r2) =>
+				new Date(r1.created_at).getDate() - new Date(r2.created_at).getDate(),
+		)
+	}
+}
+
+// GitHub API currently returns a 500 HTTP response if you attempt to fetch over 1000 releases.
+const RELEASE_COUNT_LIMIT = 1000
+
+export const findReleases = async ({
+	context,
+	targetCommitish,
+	filterByCommitish,
+	tagPrefix,
+}: {
+	context: ReleaseDrafterContext
+	targetCommitish: string
+	filterByCommitish: boolean
+	tagPrefix: string
+}) => {
+	let releaseCount = 0
+	const releases = (await context.octokit.paginate(
+		context.octokit.rest.repos.listReleases.endpoint.merge({
+			owner: context.owner,
+			repo: context.repo,
+			per_page: 100,
+		}),
+		(response, done) => {
+			releaseCount += response.data.length
+			if (releaseCount >= RELEASE_COUNT_LIMIT) {
+				done()
+			}
+			return response.data
+		},
+	)) as GitHubRelease[]
+
+	// log({ context, message: `Found ${releases.length} releases` })
+
+	// `refs/heads/branch` and `branch` are the same thing in this context
+	const headReferenceRegex = /^refs\/heads\//
+	const targetCommitishName = targetCommitish.replace(headReferenceRegex, '')
+	const commitishFilteredReleases = filterByCommitish
+		? releases.filter(
+				(r) =>
+					targetCommitishName ===
+					r.target_commitish.replace(headReferenceRegex, ''),
+		  )
+		: releases
+	const filteredReleases = tagPrefix
+		? commitishFilteredReleases.filter((r) => r.tag_name.startsWith(tagPrefix))
+		: commitishFilteredReleases
+	const sortedPublishedReleases = sortReleases(
+		filteredReleases.filter((r) => !r.draft),
+	)
+	const draftRelease = filteredReleases.find((r) => r.draft)
+	const lastRelease =
+		sortedPublishedReleases[sortedPublishedReleases.length - 1]
+
+	// TODO(jetersen): Move to outer methods
```

Comment:
Comment to be updated?

Reply:
The todo can be removed it is done.

However I plan to refactor the code some more.
I think I might have to deal with logging in core but in a better way.

I don't want to use Probot in `./packages/app` because it has too many deps and cannot update fast enough.
They are still stuck on Pino v6. Also the app only uses webhooks, so octokit webhooks should suffice.

Also I don't want to use Pino, it adds too much coding over my head imho.

GitHub action has `core.info` and `core.warning` which maps to `process.stdout`

I want to explore the refactoring options.